libvast/builtins/operators/write_to.cpp

```diff
@@ -0,0 +1,327 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/arrow_table_slice.hpp>
+#include <vast/concept/convertible/data.hpp>
+#include <vast/concept/convertible/to.hpp>
+#include <vast/concept/parseable/vast/pipeline.hpp>
+#include <vast/element_type.hpp>
+#include <vast/error.hpp>
+#include <vast/pipeline.hpp>
+#include <vast/plugin.hpp>
+#include <vast/type.hpp>
+
+#include <arrow/type.h>
+#include <caf/expected.hpp>
+
+#include <algorithm>
+#include <utility>
+
+namespace vast::plugins::write_to {
+
+namespace {
+
+/// The operator for printing data that will have to be joined later
+/// during pipeline execution.
+class print_operator final : public crtp_operator<print_operator> {
+public:
+  explicit print_operator(const printer_plugin& printer) noexcept
+    : printer_plugin_{&printer} {
+  }
+
+  auto operator()(generator<table_slice> input, const type& input_schema,
+                  operator_control_plane& ctrl) const
```

Comment:
The new API does not take a schema anymore, you can remove this parameter.

Btw, if `make_printer` should stay schema specific, you can probably use `schematic_operator<print_operator, printer_plugin::printer, chunk_ptr>` here.

Reply:
We realigned the API.